File: app-api/src/main/java/be/insaneprogramming/cleanarch/boundary/ListBuildings.java
package be.insaneprogramming.cleanarch.boundary;

import java.util.function.Consumer;

import be.insaneprogramming.cleanarch.requestmodel.ListBuildingsRequest;
import be.insaneprogramming.cleanarch.responsemodel.BuildingResponseModel;

@FunctionalInterface
@Boundary
public interface ListBuildings {
	void execute(ListBuildingsRequest request, Consumer<BuildingResponseModel> responseModelFunction);
}

File: app-api/src/main/kotlin/be/insaneprogramming/cleanarch/boundary/ListBuildings.kt
package be.insaneprogramming.cleanarch.boundary

import java.util.concurrent.CompletableFuture

import be.insaneprogramming.cleanarch.requestmodel.ListBuildingsRequest

@FunctionalInterface
interface ListBuildings {
    fun <T> execute(request: ListBuildingsRequest, buildingListPresenter: BuildingListPresenter<T>): CompletableFuture<T>
}
